### PR TITLE
[SPARK-20600][SS] KafkaRelation should be pretty printed in web UI

### PR DESCRIPTION
**Pull request by jaceklaskowski**
## What changes were proposed in this pull request?

User-friendly name of `KafkaRelation` in web UI (under Details for Query).

### Before

spark-20600-before

### After

spark-20600-after

## How was this patch tested?

Local build

```
./bin/spark-shell --jars ~/.m2/repository/org/apache/spark/spark-sql-kafka-0-10_2.11/2.3.0-SNAPSHOT/spark-sql-kafka-0-10_2.11-2.3.0-SNAPSHOT.jar --packages org.apache.kafka:kafka-clients:0.10.0.1
```

### CODE REVIEW

File: external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaRelation.scala [hunk 1]
```diff
@@ -143,4 +143,6 @@ private[kafka010] class KafkaRelation(
         validateTopicPartitions(partitions, partitionOffsets)
     }
   }
+
+  override def toString: String = "kafka"
 }
```
Comment (uncleGen) [inline external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaRelation.scala]: How about giving some more information about the kafka source? like topic, partition? refers to https://github.com/jaceklaskowski/spark/blob/2ffe4476553cfe50eb6392d8e573545a92fef737/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRelation.scala#L140

Comment (jaceklaskowski) [inline external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaRelation.scala]: Hehe...been thinking about it too, but thought it'd be too much for such a simple change.

It merits a JIRA issue though where one had to find out what and how to display. After all, you could read from multiple topics with different Kafka brokers and offsets. I was worried it'd be too much to handle as part of this issue.

I'm up for working on the other issue, but don't think we should hold this one up. WDYT?

Comment (zsxwing) [inline external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaRelation.scala]: How about this? Just show the most important parameters like `s"KafkaRelation(strategy=$strategy, start=$startingOffsets, end=$endingOffsets)"`?

**Comment by jaceklaskowski** [comment]
spark-kafkarelation

**Comment by zsxwing** [comment]
LGTM. Merging to master and 2.2. Thanks!

**Pull request closed without merging**